### PR TITLE
[IMP] report_qweb_element_page_visibility: Also be able to use the class in the report body

### PR DESCRIPTION
**Pull request by thomaspaulb**
Not only in header and footer of the report.

[subst is not called in body](https://github.com/OCA/OCB/blob/8.0/addons/report/models/report.py#L256)

### CODE REVIEW

**Comment by thomaspaulb** [comment]
@hbrunn If you can, please review.

File: report_qweb_element_page_visibility/views/layouts.xml [hunk 1]
```diff
@@ -1,48 +1,50 @@
...
+                <attribute name="onload">subst_wrapper();</attribute>
+            </xpath>
+            <xpath expr="//t[@t-if='subst is True']" position="after">
+                <script>
+                    function subst_wrapper() {
+                        if (typeof subst === "function") {
+                            subst();
+                        }
+                        var vars = {};
+                        var x = document.location.search.substring(1).split('&amp;amp;');
```
Comment (hbrunn) [inline report_qweb_element_page_visibility/views/layouts.xml]: you can avoid this kind of stuff by wrapping all javascript code in a [CDATA](https://en.wikipedia.org/wiki/CDATA) section

**Review by hbrunn — APPROVED** [review]
refactoring looks good. But it seems I'm overlooking something, your PR title looks like you add functionality, but I don't see this?

File: report_qweb_element_page_visibility/views/layouts.xml (hunk 1 above)
Comment (thomaspaulb) [reply report_qweb_element_page_visibility/views/layouts.xml]: Unfortunately, that made "wkhtmltopdf" not recognize the Javascript function anymore and thus not calling it.

**Comment by thomaspaulb** [comment]
@hbrunn The added functionality is that "subst_wrapper()" is called for header, body and footer, whereas the before modification of "subst()" was only called for header and footer. (Because the subst function is included in the HTML only when subst=True, and subst=False for body rendering, see also the code link referenced above).

**Comment by hbrunn** [comment]
ah, thanks. Positive review stands

**Comment by pedrobaeza** [comment]
Can you check Travis and runbot status?

**Comment by thomaspaulb** [comment]
Rebased. Seems that Travis/runbot errors are not mine

**Comment by silverf0x0** [comment]
@thomaspaulb Hi Thomas, I was looking for this kind of improvement, as I really need to show elements in the body report, rather then header and/or footer. I've tested your changes, but unfortunately it still show the element only in the header or in the footer.

**Comment by silverf0x0** [comment]
@hbrunn You have given positive review to this commit, which means it works. Am I missing something to make it work in the body of my report?

**Comment by Rad0van** [comment]
@thomaspaulb nice addition. Why didn't this go through? Is it worth adapting to 12.0/13.0? Thanx.

**Pull request closed without merging**